deep_quoridor/src/agents/alphazero/nn_evaluator.py

```diff
@@ -45,12 +45,16 @@ def evaluate(self, game: Quoridor):
         # Mask the policy to ignore invalid actions
         valid_actions = game.get_valid_actions()
         valid_action_indices = [self.action_encoder.action_to_index(action) for action in valid_actions]
-        policy_masked = np.zeros_like(unmasked_policy)
+        policy_masked = np.full_like(unmasked_policy, -1e32)
         policy_masked[valid_action_indices] = unmasked_policy[valid_action_indices]
 
-        # Re-normalize probabilities after masking.
-        policy_probs = policy_masked
-        policy_probs = policy_probs / policy_probs.sum()
+        if np.all(policy_masked == 0):
+            print("Policy is all zeros")
+            print(policy_masked)
+            print(game)
```

Comment:
I'm not a fan of adding debug prints to committed code - it gets out of hand fast.

Reply:
Maybe we should be using a logging tool with different levels of log that we can turn on and off, and this should be a warning

Reply:
I think that once the policy is all zeros, you're completely hosed, right? This could raise an exception.

Reply:
I'll leave the check (after the softmax now) anyway, JIC, but I posit it will never trigger.